ssl/statem/statem_clnt.c

```diff
@@ -2013,6 +2013,7 @@ static int tls_process_ske_srp(SSL *s, PACKET *pkt, EVP_PKEY **pkey)
 
 static int tls_process_ske_dhe(SSL *s, PACKET *pkt, EVP_PKEY **pkey)
 {
+#ifndef OPENSSL_NO_DH
```

Comment:
I've not reviewed all of this PR - but this change looks wrong. There is no reason why this code should be disabled in a no-dh build. It should work just fine....it just means that the provider code doesn't have dh support built in. But if you drop in a provider that has DH support, then that's fine. We recently removed these guards.

Reply:
Nope...this change is still present :-)

Reply:
Meanwhile this is fixed.